go-controller/pkg/ovn/zone_interconnect/zone_ic_handler.go

```diff
@@ -791,6 +794,69 @@ func (zic *ZoneInterconnectHandler) getStaticRoutes(ipPrefixes []*net.IPNet, nex
 	return staticRoutes
 }
 
+// deleteStaleStaticRoutes removes static routes referencing old join or transit switch
+// IPs as either IPPrefix or NextHop for a remote zone node.
+func (zic *ZoneInterconnectHandler) deleteStaleStaticRoutes(node *corev1.Node) error {
+	var v4TransitSwitchIP, v6TransitSwitchIP, v4JoinIP, v6JoinIP *net.IPNet
+
+	nodeTransitSwitchPortIPs, err := util.ParseNodeTransitSwitchPortAddrs(node)
+	if err != nil || len(nodeTransitSwitchPortIPs) == 0 {
+		return fmt.Errorf("failed to get the node transit switch port Ips : %v", err)
```

Comment:
I think it would be a better idea to aggregate the errors here instead of doing the early exit. There is still a chance to cleanup the join addr routes.